### PR TITLE
feat(filter): add label property

### PR DESCRIPTION
**Pull request by driskull**
**Related Issue:** #8076

## Summary

- add label prop to filter
- add test

### CODE REVIEW

File: packages/calcite-components/src/components/filter/filter.e2e.ts [hunk 1]
```diff
@@ -67,6 +67,15 @@ describe("calcite-filter", () => {
     expect(await input.getProperty("scale")).toBe(scale);
   });
 
+  it("honors label property", async () => {
+    const page = await newE2EPage();
+    const label = "hell world";
```
Comment (geospatialem) [inline packages/calcite-components/src/components/filter/filter.e2e.ts:72]: 😨! Did you mean to type hello 👋🏻?

**Review by geospatialem — APPROVED** [review]
LGTM! 🏆

File: packages/calcite-components/src/components/filter/filter.e2e.ts (hunk 1 above)
Comment (driskull) [reply packages/calcite-components/src/components/filter/filter.e2e.ts:72]: whoops lol

**Review by benelan — APPROVED** [review]
:+1:

**Pull request merged**